crates/sequencer/src/sequencer.rs

```diff
@@ -43,6 +49,24 @@ where
     }
 }
 
+impl<S> Sequencer<S>
+where
+    S: DumpLoad + Clone,
+    for<'any> &'any mut S: State + StateReader,
+{
+    pub fn dump_state_to_file(&self, file_path: &Path) -> Result<(), SerializationError> {
+        self.state.clone().dump_state_to_file(file_path)
```

Comment:
why do we need to clone state? isn't it big and lengthy?
maybe this method should consume self

Reply:
true !